proto/fulfillment/v1/cluster_order_type.proto

```diff
@@ -15,51 +15,157 @@ syntax = "proto3";
 
-// A cluster order contains the details that the user provides to request the provisioning of a cluster, as well as the
-// current status of the order provided by the system.
+import "fulfillment/v1/condition_status_type.proto";
+import "google/protobuf/timestamp.proto";
+
+// Contains the details that the user provides to request the provisioning of a cluster, as well as the current status
+// of the order provided by the system.
 message ClusterOrder {
-  // Unique identifier of the order. This will be automatically generated by the system when the order is placed.
+  // Unique identifier of the order.
+  //
+  // This will be automatically generated by the system when the order is placed.
   string id = 1;
 
   ClusterOrderSpec spec = 2;
   ClusterOrderStatus status = 3;
 }
 
-// The spec contains the details that the user provides to request the provisioning of the cluster.
+// Contains the details that the user provides to request the provisioning of the cluster.
 message ClusterOrderSpec {
-  // Reference to the cluster template. This is mandatory, and must be the value of the `id` field of one of the cluster
-  // templates.
+  // Reference to the cluster template.
+  //
+  // This is mandatory, and must be the value of the `id` field of one of the cluster templates.
   string template_id = 2;
 }
 
-// The status contains the current status of the order.
+// Contains the current status of the order.
 message ClusterOrderStatus {
-  // State indicates the current state of the processing of the order.
-  ClusterOrderState state = 4;
+  // Indicates the overall state of the order.
+  //
+  // For more details check the conditions.
+  ClusterOrderState state = 1;
+
+  // Contains a list of conditions that describe in detail the status of the order.
+  //
+  // For example, an order that failed could be represented like this (when converted to JSON):
+  //
+  //    {
+  //      "id": "123",
+  //      "spec": {
+  //        "template_id": "245"
+  //      },
+  //      "state": "CLUSTER_ORDER_STATE_FAILED",
+  //      "status": {
+  //        "conditions": [
+  //          {
+  //            "type: "CLUSTER_ORDER_CONDITION_TYPE_ACCEPTED",
+  //            "status": "CONDITION_STATUS_TRUE",
+  //            "last_transition_time": "2025-03-12 20:15:59+00:00",
+  //            "message": "The order has been automatically approved"
+  //          },
+  //          {
+  //            "type": "CLUSTER_ORDER_CONDITION_TYPE_FULFILLED",
+  //            "status": "CONDITION_STATUS_FALSE",
+  //            "last_transition_time": "2025-03-12 20:17:16+00:00"
+  //          },
+  //          {
+  //            "type": "CLUSTER_ORDER_CONDITION_TYPE_FAILED",
+  //            "status": "CONDITION_STATUS_TRUE",
+  //            "last_transition_time": "2025-03-12 20:18:59+00:00",
+  //            "reason": "DnsProvisioningFailure",
+  //            "message": "Failed to create DNS domain 'example.com' because it already exists"
+  //          }
```

Comment:
Do we need both Fulfilled and Failed? Or is Failed equivalent to `Fullfilled: False`, with `Reason` providing the failure reason, etc.

Reply:
I'd say that `Fulfilled: False` only means that the fulfillment isn't complete. It may be because the order was rejected, or because it failed, or because it is still in progress.